department-of-veterans-affairs/vets-website
# MHV43497 Add-TXT-Download-Functionality-For-Pathology-Reports

## Summary
Enhanced the TXT functionality within PathologyDetails, allowing for the convenient download of a TXT file displaying the LR SURGICAL PATHOLOGY REPORT. Additionally, carried out a refactoring of the PDF component for improved performance.

[MHV-43497](https://jira.devops.va.gov/browse/MHV-43497)- Add TXT Download functionality for Pathology Reports

User Story: As a Medical Records User, I want to download my labs and test information in TXT format so I have it on my computer.

GIVEN:
WHEN:
THEN:

Feature Flag Y/N ? N
Manual Testing Y/N ? Y
Automated Testing Y/N ? Y
Accessibility Testing Y/N ? Y
UCD Validation Y/N Y

Acceptance Criteria:
AC1 Download Pathology has been added (TBD)
AC2 No significant formatting is needed until designs are final
AC3 Product a TXT with placeholder data
AC4 Formatting of TXT will occur in future story when format is finalized

AC5 Follow naming convention format

Links to UCD:
Sketch Link:
Mobile: https://www.sketch.com/s/f6118ebb-966d-4f83-afba-f9e987b65eac/a/zx9z7Ze
Desktop: https://www.sketch.com/s/f6118ebb-966d-4f83-afba-f9e987b65eac/a/ZOGvMRa
Other Design Notes

Architectural Notes: Always check for the latest designs in Sketch!

## Testing done
Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

<img width="1512" alt="Screenshot 2023-11-16 at 2 58 33 PM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/5ab1aa72-0f37-40ac-92e2-b0e9f7cca8c3">

![Screenshot 2023-11-16 at 2 58 46 PM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/8da36991-35e2-4298-b15d-a8c0227ca88d)

![Screenshot 2023-11-16 at 3 00 22 PM (2)](https://github.com/department-of-veterans-affairs/vets-website/assets/8038703/636774c2-155b-4b04-9639-4664bcce1169)




## What areas of the site does it impact?
The modification affects the PathologyDetail section of the Medical Records page.

## Acceptance criteria

### Quality Assurance & Testing


### Authentication
